sqldb/migrations.go

```diff
@@ -216,3 +244,65 @@ func (t *replacerFile) Close() error {
 	// instance, so there's nothing to do for us here.
 	return nil
 }
+
+// ApplyMigrations applies the provided migrations to the database in sequence.
+// It ensures migrations are executed in the correct order, applying both custom
+// migration functions and SQL migrations as needed. If no custom migrations are
+// specified, all SQL migrations are applied up to the latest version.
+func ApplyMigrations(db *BaseDB, migrator MigrationExecutor,
+	migrations []MigrationConfig) error {
+
+	// Sort migrations by version to ensure they are applied in order.
+	sort.SliceStable(migrations, func(i, j int) bool {
+		return migrations[i].Version < migrations[j].Version
+	})
```

Comment:
slightly scary to me cause I think this doesnt account for the case where the versions are equal. I think maybe we should have an explicit order for these code-level migrations

Reply:
basically i think it would be cool if there was an overall, explicit version for each migration as one day these can diverge quite a bit but there will always be 1 single absolute DB version that we are talking about then. Thinking like a 1:1 map from: Overall Version to migration:

`map[OverallVersionNum] -> Migration` 

where `Migrations` has fields: `type = sql/code` and then a `versionNum` where that versionNum is the sql level version number or code level version number. We can persist this overall version and use it to know where to start from.

Reply:
Thank you for the great suggestion! :) Instead of using a map, I opted for a slice, which inherently implies a predefined order. We'll still sort it before applying the migrations though.